### ISSUE TITLE
False positive for let_var_whitespace and discouraged_optional_boolean

### ISSUE THREAD

**Issue by SasibalaR** [realm/SwiftLint#2692]
New Issue Checklist

 Updated SwiftLint to the latest version 0.31.0 from 0.27.0
 I searched for existing GitHub issues

Describe the bug

Got 4 type of issues.
With Swift4.2 Xcode 10.1 , SwiftLint version 0.27.0,  using command SwiftLint lint --strict , got zero warnings.

1.redundant_objc_attribute
 though not declared as @objc class having  syntax as
@objc
@IBAction private func ... 
or 
@IBOutlet @objc dynamic private weak var textBar: UIView! throws redundant_objc_attribute warning
2. unused_setter_value
 though the variable is used outside after set call.
3. superfluous_disable_command
     var isText: Bool? // swiftlint:disable:this discouraged_optional_boolean ..

Migrated to latest SwiftLint using package installation
Tried with running in Xcode 10.2 swift5.
In addition to above 3  got 
let_var_whitespace:
if I have variable initialised, then the next statement throws  let_var_whitespace warning
 private let textString = "test lint"
 private var subString: NSAttributedString?

Environment

SwiftLint version: 0.31.0
Installation method used: package
Irrelevant of configuration file
Xcode 10.2 (Swift 5)  and Xcode 10.1( Swift 4.2)

**Comment by marcelofabri** [comment]
```
@objc
@IBAction private func ...
or
@IBOutlet @objc dynamic private weak var textBar: UIView! throws redundant_objc_attribute warning
```

those are correct

> 2. unused_setter_value
> though the variable is used outside after set call.

This is complaining that `newValue` is not used inside the setter. Can you share an example?

> superfluous_disable_command
> var isText: Bool? // swiftlint:disable:this discouraged_optional_boolean

Is `discouraged_optional_boolean` enabled in your configuration file? Please post it.

> 4. let_var_whitespace
> if I have variable initialised, then the next statement throws let_var_whitespace warning

This one might be an actual issue.

**Comment by SasibalaR** [comment]
> > superfluous_disable_command
> > var isText: Bool? // swiftlint:disable:this discouraged_optional_boolean
> 
> Is `discouraged_optional_boolean` enabled in your configuration file? Please post it.
 Yes. discouraged_optional_boolean is  enabled  configuration file.

**Comment by jpsim** [comment]
I'm sorry you're running into some potential SwiftLint bugs. Do you think you could give us enough information to reproduce any of these so we can either fix them or at least diagnose why they appear to be false positives?

**Comment by SasibalaR** [comment]
> > superfluous_disable_command
> > var isText: Bool? // swiftlint:disable:this discouraged_optional_boolean
> 
> Is `discouraged_optional_boolean` enabled in your configuration file? Please post it.
> Yes. discouraged_optional_boolean is enabled in our configuration file.

This is issue with coding . no need to use ? for Bool.
Means use var isText: Bool instead of var isText: Bool?

**Comment by SasibalaR** [comment]
Updated issue description as per latest investigation.  Currently out of 4 initially mentioned only let_var_whitespace looks issue.
This issue also, If I remove space before and after of = then, it is  not reporting.
private let textString = "test lint" --  Variable Declaration Whitespace Violation: Let and var should be separated from other statements by a blank line. (let_var_whitespace)
private let textString="test lint"

**Comment by SasibalaR** [comment]
How you are considering @objc and @objcMembers. 
@objc
final class B.... 
 equals?

@objcMembers
final class B....
